enhancements/agent-installer/simplified-registry-less-cluster-operations.md

```diff
@@ -0,0 +1,426 @@
+
+* The `spec.releases` will be used to configure the desired release contents.
+  Adding (or removing) an entry from the list will trigger an update on all the
+  control plane nodes.
+* The `spec.enabled` will be used to turn on / off the feature. In particular,
+  when turned off, all the registry services running on the control plane nodes
+  will be stopped.
+* The `status.releases` field will report the currently managed releases
+* The `status.conditions` field will be used to keep track of the activity
+  performed on the various nodes.
+
+### Topology Considerations
+
+Will function identically in all topologies.
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### Extended RHCOS ISO build
+
+The release image and operators images will be stored inside the extended RHCOS
+ISO in the storage format used by the [distribution/distribution](https://github.com/distribution/distribution)
+registry’s filesystem storage plugin.
+
+#### Distribution/distribution registry OCP integration
+
+The [distribution/distribution](https://github.com/distribution/distribution)
+registry will be used to run the registry services on each control plane
+node. The binary will be built and included in the [openshift/image-registry](https://github.com/openshift/image-registry)
+repo image, so that it will be available within an OCP release image payload.
+
+#### Installation - bootstrap phase
+
+![alt bootstrap phase](simplified-operations-bootstrap.png)
+
+Initially the registry image will be placed into a read-only [additional image
+store](https://www.redhat.com/en/blog/image-stores-podman), configured to be
+used by podman in `/etc/containers/storage.conf`.
+An initial service (added by the Appliance builder in the ISO ignition) will
+startup the registry at the very beginning of the boot phase. 
+The `/etc/containers/registries.conf` will be modified to ensure that the
+images will be pulled from the local registry, using `localhost` as the
+registry host.
+
+#### Installation - install phase
+
+![alt install phase](simplified-operations-install.png)
+
+A systemd service injected by MCO will ensure the registry will be available
+after the reboot. 
+The registry will have to run on a host network port on each node to ensure
+it's accessible from the outside, specifically port 22625 (to be [registered](https://github.com/openshift/enhancements/blob/master/dev-guide/host-port-registry.md)). 
+An Assisted Service pre-flight validation will ensure that the port is open.
+
+#### Upgrade
+
+A udev rule, along with a systemd service injected by MCO will detect when an
+extended RHCOS ISO is attached, an then mount it to launch a temporary registry
+to serve its content.
+After the user modified the IRI resource with the new release pullspec, the new
+OCP release image payload will be copied on each node registry using skopeo.
```

Comment:
Skopeo is pointed directly at the host service or does it also get an IMDS entry and skopeo thinks it's talking to quay.io? If it had an IMDS entry could the IRI controller be used, unaltered, in a connected environment because it would eventually synchronize from the origin?

Reply:
There's something to be said for that, and also something to be said for not doing that because users who are connected but bandwidth-constrained will hate us if we download data over the network when they already supplied it locally :smile:

Either way, the answer needs to be made explicit in this proposal.

Reply:
They'd have supplied it locally via an IMDS pointed at their local registry, right? I assume there's a way to say never even attempt to reach the origin and fail instead?